src/ServiceControl/Operations/RetryConfirmationProcessor.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace ServiceControl.Operations
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Contracts.MessageFailures;
+    using Infrastructure.DomainEvents;
+    using MessageFailures;
+    using NServiceBus.Transport;
+    using Raven.Abstractions.Commands;
+    using Raven.Abstractions.Data;
+    using Recoverability;
+
+    class RetryConfirmationProcessor
+    {
+        public const string SuccessfulRetryHeader = "ServiceControl.Retry.Successful";
+        const string RetryUniqueMessageIdHeader = "ServiceControl.Retry.UniqueMessageId";
+
+        public RetryConfirmationProcessor(IDomainEvents domainEvents)
+        {
+            this.domainEvents = domainEvents;
+        }
+
+        public IReadOnlyCollection<ICommandData> Process(List<MessageContext> contexts)
+        {
+            var allCommands = new List<ICommandData>(contexts.Count);
```

Comment:
that list would be too small by default given we return two commands per context?

Reply:
:+1: Fixed now